src/Features/LanguageServer/ProtocolUnitTests/Initialize/LocaleTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.LanguageServer.Handler;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.UnitTests;
+
+public class LocaleTests(ITestOutputHelper? testOutputHelper) : AbstractLanguageServerProtocolTests(testOutputHelper)
+{
+    protected override TestComposition Composition => base.Composition
+            .AddParts(typeof(LocaleTestHandler));
+
+    [Theory, CombinatorialData]
+    public async Task TestUsesLspLocale(bool mutatingLspWorkspace)
+    {
+        await using var testLspServer = await CreateTestLspServerAsync(string.Empty, mutatingLspWorkspace, new InitializationOptions
+        {
+            Locale = "ja"
+        });
+
+        var result = await testLspServer.ExecuteRequestAsync<Request, Response>(LocaleTestHandler.MethodName, new Request(), CancellationToken.None);
+        Assert.Equal("ja-JP", result!.HandlerCulture);
+    }
+
+    [Theory, CombinatorialData]
+    public async Task TestUsesLspLocalePerServer(bool mutatingLspWorkspace)
+    {
+        await using var testLspServerOne = await CreateTestLspServerAsync(string.Empty, mutatingLspWorkspace, new InitializationOptions
+        {
+            Locale = "ja"
+        });
+
+        await using var testLspServerTwo = await CreateTestLspServerAsync(string.Empty, mutatingLspWorkspace, new InitializationOptions
+        {
+            Locale = "zh"
+        });
+
+        var resultOne = await testLspServerOne.ExecuteRequestAsync<Request, Response>(LocaleTestHandler.MethodName, new Request(), CancellationToken.None);
+        var resultTwo = await testLspServerTwo.ExecuteRequestAsync<Request, Response>(LocaleTestHandler.MethodName, new Request(), CancellationToken.None);
+        Assert.Equal("ja-JP", resultOne!.HandlerCulture);
+        Assert.Equal("zh-CN", resultTwo!.HandlerCulture);
+    }
+
+    [Theory, CombinatorialData]
+    public async Task TestUsesDefaultLocaleIfNotProvided(bool mutatingLspWorkspace)
+    {
+        var currentCulture = CultureInfo.CurrentUICulture.Name;
+        await using var testLspServer = await CreateTestLspServerAsync(string.Empty, mutatingLspWorkspace, new InitializationOptions
+        {
+            Locale = null
+        });
+
+        var result = await testLspServer.ExecuteRequestAsync<Request, Response>(LocaleTestHandler.MethodName, new Request(), CancellationToken.None);
+        Assert.Equal(currentCulture, result!.HandlerCulture);
+    }
+
```

Comment:
Did you want a test for a broken culture string, since you had an error handler for it?

Reply:
added